Size is not correctly returned after upload

I'm using the returned size to double-check the upload as we found some of our uploads were truncated and are trying to work out why. This works with GCS but breaks our tests. You can also see this by adding an additional assertion to upload_test.go:

```
$ git diff
diff --git a/fakestorage/upload_test.go b/fakestorage/upload_test.go
index 6a9650e..d6bd30e 100644
--- a/fakestorage/upload_test.go
+++ b/fakestorage/upload_test.go
@@ -82,6 +82,9 @@ func TestServerClientObjectWriter(t *testing.T) {
                                                n, baseContent)
                                }
 
+                               if returnedSize := w.Attrs().Size; returnedSize != int64(len(content)) {
+                                       t.Errorf("wrong writer.Attrs() size returned\nwant %d\ngot  %d", len(content), returnedSize)
+                               }
                                if returnedChecksum := w.Attrs().CRC32C; returnedChecksum != u32Checksum {
                                        t.Errorf("wrong writer.Attrs() checksum returned\nwant %d\ngot  %d", u32Checksum, returnedChecksum)
                                }
$ go test github.com/fsouza/fake-gcs-server/fakestorage
--- FAIL: TestServerClientObjectWriter (0.02s)
    --- FAIL: TestServerClientObjectWriter/http_listener (0.24s)
        --- FAIL: TestServerClientObjectWriter/http_listener/default_chunk_size (0.07s)
            upload_test.go:86: wrong writer.Attrs() size returned
                want 4718592
                got  0
        --- FAIL: TestServerClientObjectWriter/http_listener/small_chunk_size (0.17s)
            upload_test.go:86: wrong writer.Attrs() size returned
                want 4718592
                got  0
    --- FAIL: TestServerClientObjectWriter/https_listener (0.25s)
        --- FAIL: TestServerClientObjectWriter/https_listener/default_chunk_size (0.08s)
            upload_test.go:86: wrong writer.Attrs() size returned
                want 4718592
                got  0
        --- FAIL: TestServerClientObjectWriter/https_listener/small_chunk_size (0.16s)
            upload_test.go:86: wrong writer.Attrs() size returned
                want 4718592
                got  0
    --- FAIL: TestServerClientObjectWriter/no_listener (0.25s)
        --- FAIL: TestServerClientObjectWriter/no_listener/default_chunk_size (0.08s)
            upload_test.go:86: wrong writer.Attrs() size returned
                want 4718592
                got  0
        --- FAIL: TestServerClientObjectWriter/no_listener/small_chunk_size (0.16s)
            upload_test.go:86: wrong writer.Attrs() size returned
                want 4718592
                got  0
FAIL
FAIL    github.com/fsouza/fake-gcs-server/fakestorage   0.806s
FAIL
```

I looked at sending a PR, but while it seems obvious how to set for most cases where ObjectAttrs{} is created when Content is already available, it wasn't obvious how to set it for a resumable upload.

--- a/fakestorage/object.go
+++ b/fakestorage/object.go
@@ -59,7 +59,7 @@ func (o Object) MarshalJSON() ([]byte, error) {
 	temp := struct {
 		BucketName      string            `json:"bucket"`
 		Name            string            `json:"name"`
-		Size            int64             `json:"-"`
+		Size            int64             `json:"size,string"`
 		ContentType     string            `json:"contentType"`
 		ContentEncoding string            `json:"contentEncoding"`
 		Content         []byte            `json:"-"`
@@ -76,6 +76,7 @@ func (o Object) MarshalJSON() ([]byte, error) {
 		Name:            o.Name,
 		ContentType:     o.ContentType,
 		ContentEncoding: o.ContentEncoding,
+		Size:            o.Size,
 		Content:         o.Content,
 		Crc32c:          o.Crc32c,
 		Md5Hash:         o.Md5Hash,
@@ -97,7 +98,7 @@ func (o *Object) UnmarshalJSON(data []byte) error {
 	temp := struct {
 		BucketName      string            `json:"bucket"`
 		Name            string            `json:"name"`
-		Size            int64             `json:"-"`
+		Size            int64             `json:"size,string"`
 		ContentType     string            `json:"contentType"`
 		ContentEncoding string            `json:"contentEncoding"`
 		Content         []byte            `json:"-"`
@@ -117,6 +118,7 @@ func (o *Object) UnmarshalJSON(data []byte) error {
 	o.Name = temp.Name
 	o.ContentType = temp.ContentType
 	o.ContentEncoding = temp.ContentEncoding
+	o.Size = temp.Size
 	o.Content = temp.Content
 	o.Crc32c = temp.Crc32c
 	o.Md5Hash = temp.Md5Hash

--- a/fakestorage/upload_test.go
+++ b/fakestorage/upload_test.go
@@ -82,6 +82,9 @@ func TestServerClientObjectWriter(t *testing.T) {
 						n, baseContent)
 				}
 
+				if returnedSize := w.Attrs().Size; returnedSize != int64(len(content)) {
+					t.Errorf("wrong writer.Attrs() size returned\nwant %d\ngot  %d", len(content), returnedSize)
+				}
 				if returnedChecksum := w.Attrs().CRC32C; returnedChecksum != u32Checksum {
 					t.Errorf("wrong writer.Attrs() checksum returned\nwant %d\ngot  %d", u32Checksum, returnedChecksum)
 				}

